mcp/src/main/java/io/modelcontextprotocol/client/McpAsyncClient.java

```diff
@@ -335,6 +346,26 @@ public Mono<Void> closeGracefully() {
 		return this.mcpSession.closeGracefully();
 	}
 
+	// --------------------------
+	// Utility Methods
+	// --------------------------
+
+	/**
+	 * Utility method to handle the common pattern of checking initialization before
+	 * executing an operation.
+	 * @param <T> The type of the result Mono
+	 * @param errorMessage The error message to use if the client is not initialized
+	 * @param operation The operation to execute if the client is initialized
+	 * @return A Mono that completes with the result of the operation
+	 */
+	private <T> Mono<T> withInitializationCheck(String errorMessage,
```

Comment:
"Client must be initialized before " is repeated in each use, consider replacing `errorMessage` with `action` and just specify what's about to happen, e.g. `action="pinging the server"`.